pkg/util/http.go

```diff
@@ -342,3 +343,9 @@ func copyValues(src url.Values) url.Values {
 	}
 	return dst
 }
+
+// IsHTTPStatusCode returns true if the given code is a valid HTTP status code, or false otherwise.
+func IsHTTPStatusCode(code codes.Code) bool {
+	httpStatus := http.StatusText(int(code))
+	return httpStatus != ""
+}
```

Comment:
This needs a test. I can imagine a next golang version returning "Unknown" string for unknown `http.StatusText` and breaking this.

Reply:
I will replace it with the previous implementation
```
return int(statusCode) >= 100 && int(statusCode) < 600
```